### DISCUSSION TITLE
7581/7582/7530ax:  Freetz GUI accessible again on port :81

### DISCUSSION THREAD

**Discussion by roland3939** [Freetz-NG/freetz-ng#485] [Allgemein] [answered]
When booting freezt-ng image on 7581, port 81 and port 443 are used by process `/bin/smd` ( see [issue #301 comment](https://github.com/Freetz-NG/freetz-ng/issues/301#issuecomment-1050081412))
further investigation shows,  process `smd` ( **s**ystem **m**anagement **d**aemon from Broadcom's SW stack used in AVM's  firmware) starts the busyboy  applet `httpd` ( `/usr/sbin/httpd` )  as a child process. This process chain results in port 81 no longer being available  for use by Freetz GUI.
simply terminate `smd` process by 
```
/etc/rc3.d/S65smd stop
```
is not sufficient:   e.g.  on event "WLAN status change"  a box reboot gets triggered

Possible options to fix this issue during freetz-ng image built:

* change the  `web-cfg` port  of freetz-ng (e.g. to :82)

* delete symlink for busybox's `httpd`

* compile busybox without `httpd` web-server applet

* modify configuration file used by `smd` binary
**set line comment at  lines 40 to 47** in file `/etc/cms_entity_info.d/eid_bcm_mgmt.txt`
this prevents the `smd` process  from starting the `httpd` web-server daemon
Box runs long-term stable  with this option applied.

Thus, Freetz GUI (`web-cfg`) is accessible again on  port :81 and  additionally port 443 is used  by `ctlmgr` (https access to AVM GUI ).

**Comment by fda77** [comment]
Thx for investigation! Freetz-webif uses also the httpd. 1st i thought automatic freetz-webif port change would be the best, but its then hard to find. 
Whats about changing the port of smb? `HTTPD_PORT` is defined in  `etc/cms_entity_info.d/symbol_table.txt`. This could work if the port is not used by someting else. Can you access it with browser? Is there something usefull?

If you want to (hacky) change symbol_table.txt:
- Build image as you need
- change the .txt in build/original/filesystem/.....
- Build the image again (unpacking should be skipped)

**Comment by roland3939** [comment, marked as answer]
changing `HTTPD_PORT`  and `HTTPD_PORT_SSL` is not the best option, as `httpd` is still forked as a child process by `smd` ( only using some different ports then). 
it's useless  since there  is no configuration given for this `httpd` instance.

to prevent the `smd` process from starting the `httpd` web-server daemon is a much better option, I think; and it doesn't interfere with the usage of `httpd` by Freetz-webif  not at all.
So, **set line comment at lines 40 to 47** in file `/etc/cms_entity_info.d/eid_bcm_mgmt.txt` is my preferred fix ( as already stated in https://github.com/Freetz-NG/freetz-ng/discussions/485#discussion-3906941)
for convenience,  just add attached file **`840_patch_smd_no_httpd.sh`**  to `patches/scripts` folder and the whole is finished.

[840_patch_smd_no_httpd.sh.txt](https://github.com/Freetz-NG/freetz-ng/files/8214968/840_patch_smd_no_httpd.sh.txt)

**Reply by fda77** [reply]
Thx! Fixed: c9f388e61
I've switched to patches to get notified if the file changes in some way. Just delete static line numbers could cause anyhting
